common/nodejs-utils/src/config.rs

```diff
@@ -0,0 +1,338 @@
+        .and_then(|contents| {
+            toml_edit::DocumentMut::from_str(&contents).map_err(ConfigError::ParseProjectToml)
+        })?;
+
+    match get_buildpack_namespaced_config(&project_toml)? {
+        Some(namespaced_config) => match namespaced_config.get(PRUNE_DEV_DEPENDENCY_CONFIG) {
+            Some(config) => config.as_bool().map(Some).ok_or(ConfigError::WrongType),
+            None => Ok(None),
+        },
+        None => Ok(None),
+    }
+}
+
+fn get_buildpack_namespaced_config(
+    doc: &toml_edit::DocumentMut,
+) -> Result<Option<&dyn TableLike>, ConfigError> {
+    let mut current_table = doc
+        .as_item()
+        .as_table_like()
+        .ok_or(ConfigError::ExpectedTomlTable)?;
+    for name in CONFIG_NAMESPACE {
+        current_table = match current_table.get(name) {
+            Some(item) => match item.as_table_like() {
+                Some(table) => table,
+                None => return Err(ConfigError::ExpectedTomlTable),
+            },
+            None => return Ok(None),
+        };
+    }
+    Ok(Some(current_table))
+}
+
+#[derive(Debug)]
+pub enum ConfigError {
+    CheckExists(std::io::Error),
+    ReadProjectToml(std::io::Error),
+    ParseProjectToml(toml_edit::TomlError),
+    ExpectedTomlTable,
+    WrongType,
+}
+
+impl From<ConfigError> for SharedErrorMessageBuilder {
+    fn from(value: ConfigError) -> Self {
+        let project_toml = style::value("project.toml");
+        let toml_spec_url = style::url("https://toml.io/en/v1.0.0");
+        let config_table_key = style::value(format!("[{}]", CONFIG_NAMESPACE.to_vec().join(".")));
+        let config_key = style::value(PRUNE_DEV_DEPENDENCY_CONFIG);
+
+        match value {
+            ConfigError::CheckExists(e) => error_message()
+                .error_type(ErrorType::Internal)
+                .header("Failed to check if project.toml exists")
+                .body(formatdoc! { "
+                    An unexpected I/O error occurred while checking if {project_toml} exists in the \
+                    root of the application.
+                "})
+                .debug_info(e.to_string()),
+
+            ConfigError::ReadProjectToml(e) => error_message()
+                .error_type(ErrorType::Internal)
+                .header("Failed to read project.toml")
+                .body(formatdoc! { "
+                    This buildpack will read from {project_toml} if the file is present in the root of \
+                    the application but an unexpected I/O error occurred during this operation.
+                "})
+                .debug_info(e.to_string()),
+
+            ConfigError::ParseProjectToml(e) => error_message()
+                .error_type(UserFacing(SuggestRetryBuild::Yes, SuggestSubmitIssue::No))
+                .header("Failed to parse project.toml")
+                .body(formatdoc! { "
+                    This buildpack reads configuration from {project_toml} to complete the build but \
```

Comment:
Grammar, missing comma:

```suggestion
                    This buildpack reads configuration from {project_toml} to complete the build, but \
```

Reply:
Fixed in bc36e48a